plugins/modules/aci_hsrp_interface_policy.py

```diff
@@ -0,0 +1,303 @@
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"], required=True),  # Not required for querying all objects
+        hsrp=dict(type="str", aliases=["hsrp_interface", "name"]),  # Not required for querying all objects
+        description=dict(type="str", aliases=["descr"]),
+        controls=dict(type="list", elements="str", choices=["bfd", "bia"]),
+        reload_delay=dict(type="int"),
+        delay=dict(type="int"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["hsrp"]],
+            ["state", "present", ["hsrp"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    tenant = module.params.get("tenant")
+    hsrp = module.params.get("hsrp")
+    description = module.params.get("description")
+    controls = ",".join(module.params.get("controls")) if module.params.get("controls") else None
+
+    reload_delay = module.params.get("reload_delay")
+    if reload_delay is not None and reload_delay not in range(1, 10000):
+        module.fail_json(msg="Parameter 'reload_delay' is only valid in range between 1 and 10000.")
+
+    delay = module.params.get("delay")
+    if delay is not None and delay not in range(1, 10000):
+        module.fail_json(msg="Parameter 'delay' is only valid in range between 1 and 10000.")
+
+    state = module.params.get("state")
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="hsrpIfPol",
+            aci_rn="tn-{0}/hsrpIfPol-{1}".format(tenant, hsrp),
+            module_object=hsrp,
+            target_filter={"name": hsrp},
```

Comment:
why not have tenant as root class?

Reply:
I included it in the RN instead. However, reverted it for consistency.